Launcher/Classes/LocalizationHandler.swift

```diff
@@ -0,0 +1,22 @@
+import Foundation
+
+class LocalizationHandler {
+    
+    let applicationStateHandler = ApplicationStateHandler()
+    
+    func getKey(forString: String) -> [String] {
+        
+        guard
+            let value = parseResponse(response: forString),
+            let filePath = applicationStateHandler.filePath?.appendingPathComponent("build/Calabash.app/Frameworks/XNGLocalizedString.framework/en.lproj/Localizable.strings"),
+            let stringsDict = NSDictionary(contentsOf: filePath) as? [String: String]
+        else { return [] }
+        
+        return stringsDict.keysForValue(value: value)
+    }
+    
+    private func parseResponse(response: String) -> String? {
+        return RegexHandler().matches(for: "\\marked:'(.*?)\\\'", in: response).last
```

Comment:
What is this actually trying to accomplish? It seems like a bit of magic to return `.last`?

Reply:
it parses strings like "UILabel marked:'Profile'" to get only text (Profile in this case). The last will be the last of matches for the regular expression that will be without special symbols